### ISSUE TITLE
how to possibly assign id value without getting selected object

### ISSUE THREAD

**Issue by nellyigiebor** [valor-software/ng2-select#705]
Hi all,

how can I possibly bind to selected "id" value while displaying "text" value 
without getting selected item as an object just like in angular2-select component.  

its not just cool doing this :-

 groupSelecdted(event) { this.dataFrm.controls["groupId"].setValue(event.id); }
 View  
<ng-select [items]="productGroup" [multiple]="false" [allowClear]="true" (selected)="groupSelecdted($event)"> </ng-select>

Thanks in advance for reponding.

**Comment by exedor** [comment]
I'm not sure I understand your question. [] allows you to bind a directive/attribute (i.e. property internal to the component) to a property of the parent component. () is the method for collecting output values and, for good reason, it uses, events to deliver them to the parent.

Are you asking about how to do two way data binding using [(ngModel)] with this component? Are you saying you want the selected item as a simple "value"? i.e. in your example above you would expect the value of "event" to be the id instead of an object containing the id? If so, that isn't always the most useful thing. Having both provides the most flexibility.

**Comment by nellyigiebor** [comment]
Hi,

I'm using a reactiveform builder and I need to directly bind the selected item value to my model "two-ways" just like how  "angular2-select" "https://www.npmjs.com/package/angular2-select"  component does it without the need to write fuction to bind this value like so
selectedGroupValue(event) { 
     this.dataFrm.controls["groupId"].setValue(event.id);
 }
Can you help?

Thanks for replying

**Comment by timaxapa** [comment]
Yes, @nellyigiebor you are right. It will be more comfortable to bind `id` field directly without returning complex object. I use reactive forms too and got the same issue with it.

**Comment by labyed** [comment]
+1
